### PR TITLE
[degiro]  Adding support for depot statements based on #1742

### LINKED ISSUE #1705: [Degiro] Die gesamte Datei kann nicht importiert werden.
## Document
<details><summary>Transactions202011.pdf</summary><p>

```json
[
  [
    "Xxxx Xxxxxx Xxxxxx Xxxx",
    "Xxxxxxxxxxxxxxxx 0",
    "00000 Xxxxxxxx",
    "Germany",
    "Benutzername:",
    "****xx0",
    "(00000000)",
    "Transaktionsübersicht von 01-11-2020 bis 30-11-2020",
    "Datum",
    "Uhrzei",
    "t",
    "Produkt",
    "ISIN",
    "Referen",
    "zbörse",
    "Ausführungso",
    "rt",
    "Anzahl",
    "Kurs",
    "Wert in",
    "Lokalwährung",
    "Wert",
    "Wechse",
    "lkurs",
    "Transaktions",
    "kosten",
    "Gesamt",
    "23-11-2020",
    "12:13",
    "CALL 16.12.21 ASMLHOLD 340",
    "DE000KA5U0Z1",
    "FRA",
    "FRAB",
    "307",
    "5,20",
    "EUR",
    "-1.596,40",
    "EUR",
    "-1.596,40",
    "EUR",
    "-3,76",
    "EUR",
    "-1.600,16",
    "EUR",
    "18-11-2020",
    "17:00",
    "CALL 13.12.23 DAX 13400",
    "DE000SR0JJM5",
    "FRA",
    "FRAB",
    "26",
    "19,22",
    "EUR",
    "-499,72",
    "EUR",
    "-499,72",
    "EUR",
    "-2,55",
    "EUR",
    "-502,27",
    "EUR",
    "18-11-2020",
    "16:54",
    "CALL 21.01.22 TESLA 460",
    "DE000GF525S6",
    "FRA",
    "FRAB",
    "880",
    "1,12",
    "EUR",
    "-985,60",
    "EUR",
    "-985,60",
    "EUR",
    "-3,08",
    "EUR",
    "-988,68",
    "EUR",
    "18-11-2020",
    "16:46",
    "CALL 19.11.21 AMAZON 3200",
    "DE000GF6CTR3",
    "FRA",
    "FRAB",
    "-500",
    "3,62",
    "EUR",
    "1.810,00",
    "EUR",
    "1.810,00",
    "EUR",
    "-3,99",
    "EUR",
    "1.806,01",
    "EUR",
    "18-11-2020",
    "11:55",
    "CALL 17.03.22 MICROSO. 230",
    "DE000KB5KYK1",
    "FRA",
    "FRAB",
    "1000",
    "2,03",
    "EUR",
    "-2.030,00",
    "EUR",
    "-2.030,00",
    "EUR",
    "-4,23",
    "EUR",
    "-2.034,23",
    "EUR",
    "18-11-2020",
    "11:51",
    "CALL 18.03.22 ALPHAB.C 1750",
    "DE000GF6KFA1",
    "FRA",
    "FRAB",
    "905",
    "2,20",
    "EUR",
    "-1.991,00",
    "EUR",
    "-1.991,00",
    "EUR",
    "-4,19",
    "EUR",
    "-1.995,19",
    "EUR",
    "18-11-2020",
    "11:37",
    "CALL 19.11.21 AMAZON 3200",
    "DE000GF6CTR3",
    "FRA",
    "FRAB",
    "515",
    "3,82",
    "EUR",
    "-1.967,30",
    "EUR",
    "-1.967,30",
    "EUR",
    "-4,16",
    "EUR",
    "-1.971,46",
    "EUR",
    "18-11-2020",
    "11:35",
    "CALL 16.03.22 NIKE 125",
    "DE000TT3RDK6",
    "FRA",
    "FRAB",
    "1000",
    "2,00",
    "EUR",
    "-2.000,00",
    "EUR",
    "-2.000,00",
    "EUR",
    "-4,20",
    "EUR",
    "-2.004,20",
    "EUR",
    "18-11-2020",
    "11:19",
    "CALL 19.01.22 ALIBABA 260",
    "DE000TR7MWB9",
    "FRA",
    "FRAB",
    "243",
    "4,07",
    "EUR",
    "-989,01",
    "EUR",
    "-989,01",
    "EUR",
    "-3,09",
    "EUR",
    "-992,10",
    "EUR",
    "18-11-2020",
    "10:03",
    "CALL 19.11.21 AMAZON 3200",
    "DE000GF6CTR3",
    "FRA",
    "FRAB",
    "500",
    "3,78",
    "EUR",
    "-1.890,00",
    "EUR",
    "-1.890,00",
    "EUR",
    "-4,08",
    "EUR",
    "-1.894,08",
    "EUR",
    "18-11-2020",
    "09:43",
    "CALL 17.06.21 ADIDAS 282",
    "DE000CP95H65",
    "FRA",
    "FRAB",
    "-600",
    "2,30",
    "EUR",
    "1.380,00",
    "EUR",
    "1.380,00",
    "EUR",
    "-3,52",
    "EUR",
    "1.376,48",
    "EUR",
    "18-11-2020",
    "09:41",
    "CALL 16.12.21 ASMLHOLD 340",
    "DE000KA5U0Z1",
    "FRA",
    "FRAB",
    "-342",
    "5,10",
    "EUR",
    "1.744,20",
    "EUR",
    "1.744,20",
    "EUR",
    "-3,92",
    "EUR",
    "1.740,28",
    "EUR",
    "18-11-2020",
    "09:06",
    "CALL 16.06.21 ALPHAB.C 1700",
    "DE000TR9BNC5",
    "FRA",
    "FRAB",
    "-1580",
    "1,68",
    "EUR",
    "2.654,40",
    "EUR",
    "2.654,40",
    "EUR",
    "-4,92",
    "EUR",
    "2.649,48",
    "EUR",
    "18-11-2020",
    "09:00",
    "CALL 17.12.21 AMAZON 2650",
    "DE000GB950N1",
    "FRA",
    "FRAB",
    "-190",
    "6,29",
    "EUR",
    "1.195,10",
    "EUR",
    "1.195,10",
    "EUR",
    "-3,31",
    "EUR",
    "1.191,79",
    "EUR",
    "18-11-2020",
    "09:00",
    "CALL 16.06.21 NIKE 95",
    "DE000TR8JLL5",
    "FRA",
    "FRAB",
    "-728",
    "3,22",
    "EUR",
    "2.344,16",
    "EUR",
    "2.344,16",
    "EUR",
    "-4,58",
    "EUR",
    "2.339,58",
    "EUR",
    "18-11-2020",
    "08:18",
    "CALL 17.06.21 MICROSO. 190",
    "DE000KA00QH5",
    "FRA",
    "FRAB",
    "-630",
    "2,74",
    "EUR",
    "1.726,20",
    "EUR",
    "1.726,20",
    "EUR",
    "-3,90",
    "EUR",
    "1.722,30",
    "EUR",
    "DEGIRO B.V. ist als",
    "Investmentgesellschaft bei der",
    "niederländischen",
    "Finanzaufsichtsbehörde (AFM)",
    "registriert.",
    "DEGIRO B.V.",
    "Amstelplein 1",
    "1096 HA Amsterdam",
    "E",
    "xxxxxxxxxxxxx@xxxxxx.xx",
    "I",
    "www.degiro.de",
    "Seite 1 /",
    "2",
    "Xxxxx 0000-00-00"
  ],
  [
    "Xxxx Xxxxxx Xxxxxx Xxxx",
    "Xxxxxxxxxxxxxxxx 0",
    "00000 Xxxxxxxx",
    "Germany",
    "Benutzername:",
    "****xx0",
    "(00000000)",
    "Datum",
    "Uhrzei",
    "t",
    "Produkt",
    "ISIN",
    "Referen",
    "zbörse",
    "Ausführungso",
    "rt",
    "Anzahl",
    "Kurs",
    "Wert in",
    "Lokalwährung",
    "Wert",
    "Wechse",
    "lkurs",
    "Transaktions",
    "kosten",
    "Gesamt",
    "17-11-2020",
    "14:36",
    "HVB CALL 16.12.20 THERMOFI",
    "415",
    "DE000HZ9DVL9",
    "FRA",
    "FRAB",
    "-190",
    "5,75",
    "EUR",
    "1.092,50",
    "EUR",
    "1.092,50",
    "EUR",
    "-3,20",
    "EUR",
    "1.089,30",
    "EUR",
    "05-11-2020",
    "19:06",
    "TUBULL",
    "O.ENDSERVICEN394,493534",
    "DE000GC88CE1",
    "FRA",
    "FRAB",
    "800",
    "1,23",
    "EUR",
    "-984,00",
    "EUR",
    "-984,00",
    "EUR",
    "-3,08",
    "EUR",
    "-987,08",
    "EUR",
    "03-11-2020",
    "09:51",
    "CALL 16.12.21 ASMLHOLD 340",
    "DE000KA5U0Z1",
    "FRA",
    "FRAB",
    "342",
    "2,88",
    "EUR",
    "-984,96",
    "EUR",
    "-984,96",
    "EUR",
    "-3,08",
    "EUR",
    "-988,04",
    "EUR",
    "03-11-2020",
    "09:47",
    "TUBEAR O.END",
    "GILEAD67,29776312",
    "DE000GF10W41",
    "FRA",
    "FRAB",
    "-1238",
    "0,82",
    "EUR",
    "1.015,16",
    "EUR",
    "1.015,16",
    "EUR",
    "-3,12",
    "EUR",
    "1.012,04",
    "EUR",
    "03-11-2020",
    "09:46",
    "TUBEAR O.END DAX",
    "13835,8078",
    "DE000KA9VA32",
    "FRA",
    "FRAB",
    "-10",
    "20,07",
    "EUR",
    "200,70",
    "EUR",
    "200,70",
    "EUR",
    "-2,22",
    "EUR",
    "198,48",
    "EUR",
    "03-11-2020",
    "09:00",
    "TUBEAR",
    "O.ENDRECKITT80,22795523",
    "DE000GC9KTV5",
    "FRA",
    "FRAB",
    "-625",
    "1,29",
    "EUR",
    "806,25",
    "EUR",
    "806,25",
    "EUR",
    "-2,89",
    "EUR",
    "803,36",
    "EUR",
    "29-10-2020",
    "11:20",
    "MINIS O.END DAX 11672,1",
    "DE000UE337W1",
    "FRA",
    "-33",
    "1,18",
    "EUR",
    "38,94",
    "EUR",
    "38,94",
    "EUR",
    "38,94",
    "EUR",
    "28-10-2020",
    "09:57",
    "TUBULL",
    "O.ENDMASTERC.303,301861",
    "DE000GF2AT89",
    "FRA",
    "-100",
    "0,001",
    "EUR",
    "0,10",
    "EUR",
    "0,10",
    "EUR",
    "0,10",
    "EUR",
    "DEGIRO B.V. ist als",
    "Investmentgesellschaft bei der",
    "niederländischen",
    "Finanzaufsichtsbehörde (AFM)",
    "registriert.",
    "DEGIRO B.V.",
    "Amstelplein 1",
    "1096 HA Amsterdam",
    "E",
    "xxxxxxxxxxxxx@xxxxxx.xx",
    "I",
    "www.degiro.de",
    "Seite 2 /",
    "2",
    "Xxxxx 0000-00-00"
  ]
]

```

</p></details>

## Community profile

jlnkrn@gmx.de

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #1742: [De Giro] De Giro Broker
## Document
<details><summary>Portfolio.pdf</summary><p>

```json
[
  [
    "Xxxx Xxxxx Xüxxxxxxxxx",
    "Xüxxxxx 00 X",
    "00000 Xxxxxxüxx",
    "Germany",
    "Benutzername:",
    "*********r22",
    "(31049693)",
    "Portfolioübersicht per 23-02-2021",
    "Produkt",
    "Symbol/ISIN",
    "Anzahl",
    "Schlußkurs",
    "Wert",
    "Wert in EUR",
    "CASH & CASH FUND & FTX CASH (EUR)",
    "EUR 0.33",
    "0,33",
    "AFC ENERGY PLC   LS -,001",
    "GB00B18S7B29",
    "17",
    "0,67",
    "EUR 11.41",
    "11,41",
    "DEUTSCHE LUFTHANSA AG",
    "DE0008232125",
    "5",
    "11,42",
    "EUR 57.10",
    "57,10",
    "DEUTZ AG",
    "DE0006305006",
    "10",
    "6,22",
    "EUR 62.20",
    "62,20",
    "HALO COLLECTIVE INC.",
    "CA40638K1012",
    "78",
    "0,12",
    "EUR 9.06",
    "9,06",
    "PALANTIR TECHNOLOGIES INC",
    "US69608A1088",
    "16",
    "23,40",
    "EUR 374.40",
    "374,40",
    "WACKER NEUSON SE",
    "DE000WACK012",
    "2",
    "16,67",
    "EUR 33.34",
    "33,34",
    "Gesamtportfoliowert",
    "EUR 547.84",
    "DEGIRO B.V. ist als",
    "Investmentgesellschaft bei der",
    "niederländischen",
    "Finanzaufsichtsbehörde (AFM)",
    "registriert.",
    "DEGIRO B.V.",
    "Amstelplein 1",
    "1096 HA Amsterdam",
    "E",
    "xxxxxxxxxxxxx@xxxxxx.xx",
    "I",
    "www.degiro.de",
    "Seite 1 /",
    "1",
    "Konto 2021-02-23"
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #2148: [] portfolio
## Document
<details><summary>Transactions (4).pdf</summary><p>

```json
[
  [
    "Xxxxxx Xxxxxxx Xxxxx",
    "xxx Xxxxxx 0",
    "00000 Xxxxxx Xxxxxxxxx",
    "Xxxxx",
    "Nome Utente:",
    "*****ino",
    "(000000000)",
    "Operazioni da 14-02-2018 a 16-03-2021",
    "Data",
    "Ora",
    "Prodotto",
    "ISIN",
    "Borsa di",
    "Borsa",
    "Quantit",
    "à",
    "Quotazione",
    "Valore locale",
    "Valore",
    "Tasso",
    "di",
    "Commission",
    "e",
    "Totale",
    "16-02-2021",
    "09:25",
    "NEXI",
    "IT0005366767",
    "MIL",
    "MTAA",
    "-203",
    "16,54",
    "XXX",
    "0.000,00",
    "XXX",
    "0.000,00",
    "XXX",
    "-2,18",
    "XXX",
    "3.355,44",
    "XXX",
    "03-02-2021",
    "11:45",
    "A2A",
    "IT0001233417",
    "MIL",
    "MTAA",
    "-500",
    "1,40",
    "XXX",
    "700,00",
    "XXX",
    "700,00",
    "XXX",
    "-0,85",
    "XXX",
    "699,15",
    "XXX",
    "18-01-2021",
    "14:45",
    "MARR",
    "IT0003428445",
    "MIL",
    "MTAA",
    "5",
    "17,50",
    "XXX",
    "-87,50",
    "XXX",
    "-87,50",
    "XXX",
    "-0,04",
    "XXX",
    "-87,54",
    "XXX",
    "18-01-2021",
    "14:45",
    "MARR",
    "IT0003428445",
    "MIL",
    "MTAA",
    "65",
    "17,50",
    "XXX",
    "-1.137,50",
    "XXX",
    "-1.137,50",
    "XXX",
    "-1,07",
    "XXX",
    "-1.138,57",
    "XXX",
    "28-12-2020",
    "15:34",
    "XPENG INC. AMERICAN",
    "DEPOSITARY SHARES EACH",
    "REPRES",
    "US98422D1054",
    "NSY",
    "XNYS",
    "15",
    "41,00",
    "USD",
    "-615,00",
    "USD",
    "-502,86",
    "XXX",
    "1,2218",
    "-0,55",
    "XXX",
    "-503,41",
    "XXX",
    "18-12-2020",
    "16:21",
    "SPDR RUSSELL 2000 US",
    "SMALL CAP UCITS ETF",
    "IE00BJ38QD84",
    "MIL",
    "MTAA",
    "-15",
    "45,025",
    "XXX",
    "675,38",
    "XXX",
    "675,38",
    "XXX",
    "-2,20",
    "XXX",
    "673,18",
    "XXX",
    "03-12-2020",
    "21:41",
    "PFIZER INC. COMMON ST",
    "US7170811035",
    "NSY",
    "XNYS",
    "35",
    "40,00",
    "USD",
    "-1.400,00",
    "USD",
    "-1.152,93",
    "XXX",
    "1,2131",
    "-0,62",
    "XXX",
    "-1.153,55",
    "XXX",
    "30-11-2020",
    "09:01",
    "IMMSI",
    "IT0001413837",
    "MIL",
    "MTAA",
    "-2900",
    "0,437",
    "XXX",
    "1.267,30",
    "XXX",
    "1.267,30",
    "XXX",
    "-1,13",
    "XXX",
    "1.266,17",
    "XXX",
    "08-10-2020",
    "09:07",
    "NEXI",
    "IT0005366767",
    "MIL",
    "43",
    "15,85",
    "XXX",
    "-681,55",
    "XXX",
    "-681,55",
    "XXX",
    "-0,84",
    "XXX",
    "-682,39",
    "XXX",
    "14-07-2020",
    "09:01",
    "ATLANTIA",
    "IT0003506190",
    "MIL",
    "-100",
    "11,30",
    "XXX",
    "1.130,00",
    "XXX",
    "1.130,00",
    "XXX",
    "-1,07",
    "XXX",
    "1.128,93",
    "XXX",
    "13-07-2020",
    "09:34",
    "ATLANTIA",
    "IT0003506190",
    "MIL",
    "100",
    "12,025",
    "XXX",
    "-1.202,50",
    "XXX",
    "-1.202,50",
    "XXX",
    "-1,10",
    "XXX",
    "-1.203,60",
    "XXX",
    "09-07-2020",
    "15:27",
    "ERG",
    "IT0001157020",
    "MIL",
    "-9",
    "21,12",
    "XXX",
    "190,08",
    "XXX",
    "190,08",
    "XXX",
    "-0,60",
    "XXX",
    "189,48",
    "XXX",
    "07-07-2020",
    "12:19",
    "BANCA SISTEMA",
    "IT0003173629",
    "MIL",
    "-500",
    "1,60",
    "XXX",
    "800,00",
    "XXX",
    "800,00",
    "XXX",
    "-0,90",
    "XXX",
    "799,10",
    "XXX",
    "06-07-2020",
    "09:04",
    "L&G BATTERY VALUE-CHAIN",
    "UCITS ETF",
    "IE00BF0M2Z96",
    "MIL",
    "-40",
    "8,51",
    "XXX",
    "340,40",
    "XXX",
    "340,40",
    "XXX",
    "-2,10",
    "XXX",
    "338,30",
    "XXX",
    "30-06-2020",
    "10:58",
    "IMMSI",
    "IT0001413837",
    "MIL",
    "2900",
    "0,39",
    "XXX",
    "-1.131,00",
    "XXX",
    "-1.131,00",
    "XXX",
    "-1,07",
    "XXX",
    "-1.132,07",
    "XXX",
    "29-06-2020",
    "13:58",
    "ETFS PHYSICAL GOLD",
    "JE00B1VS3770",
    "MIL",
    "15",
    "149,09",
    "XXX",
    "-2.236,35",
    "XXX",
    "-2.236,35",
    "XXX",
    "-2,67",
    "XXX",
    "-2.239,02",
    "XXX",
    "DEGIRO B.V. è registrata come",
    "Società d’Investimento presso l’Autorità",
    "dei Mercati Finanziari olandese.",
    "DEGIRO B.V.",
    "Amstelplein 1",
    "1096 HA Amsterdam",
    "E",
    "xxxxxxx@xxxxxx.xx",
    "I",
    "www.degiro.it",
    "Pagina 1 /",
    "2",
    "Estratto conto 2021-03-16"
  ],
  [
    "Xxxxxx Xxxxxxx Xxxxx",
    "xxx Xxxxxx 0",
    "00000 Xxxxxx Xxxxxxxxx",
    "Xxxxx",
    "Nome Utente:",
    "*****ino",
    "(000000000)",
    "Data",
    "Ora",
    "Prodotto",
    "ISIN",
    "Borsa di",
    "Borsa",
    "Quantit",
    "à",
    "Quotazione",
    "Valore locale",
    "Valore",
    "Tasso",
    "di",
    "Commission",
    "e",
    "Totale",
    "16-06-2020",
    "14:27",
    "FINECOBANK",
    "IT0000072170",
    "MIL",
    "MTAA",
    "-4",
    "12,13",
    "XXX",
    "48,52",
    "XXX",
    "48,52",
    "XXX",
    "-0,52",
    "XXX",
    "48,00",
    "XXX",
    "16-06-2020",
    "09:00",
    "ENEL .",
    "IT0003128367",
    "MIL",
    "MTAA",
    "-50",
    "7,51",
    "XXX",
    "375,50",
    "XXX",
    "375,50",
    "XXX",
    "-0,69",
    "XXX",
    "374,81",
    "XXX",
    "15-06-2020",
    "09:06",
    "A2A",
    "IT0001233417",
    "MIL",
    "MTAA",
    "500",
    "1,20",
    "XXX",
    "-600,00",
    "XXX",
    "-600,00",
    "XXX",
    "-0,80",
    "XXX",
    "-600,80",
    "XXX",
    "01-06-2020",
    "14:27",
    "BANCA SISTEMA",
    "IT0003173629",
    "MIL",
    "MTAA",
    "500",
    "1,40",
    "XXX",
    "-700,00",
    "XXX",
    "-700,00",
    "XXX",
    "-0,85",
    "XXX",
    "-700,85",
    "XXX",
    "13-05-2020",
    "09:01",
    "DIASORIN",
    "IT0003492391",
    "MIL",
    "MTAA",
    "-1",
    "160,00",
    "XXX",
    "160,00",
    "XXX",
    "160,00",
    "XXX",
    "-0,58",
    "XXX",
    "159,42",
    "XXX",
    "07-04-2020",
    "10:27",
    "DIASORIN",
    "IT0003492391",
    "MIL",
    "AQXE",
    "1",
    "125,60",
    "XXX",
    "-125,60",
    "XXX",
    "-125,60",
    "XXX",
    "-0,56",
    "XXX",
    "-126,16",
    "XXX",
    "24-02-2020",
    "14:18",
    "NEXI",
    "IT0005366767",
    "MIL",
    "CHIX",
    "90",
    "14,956",
    "XXX",
    "-1.346,04",
    "XXX",
    "-1.346,04",
    "XXX",
    "-1,17",
    "XXX",
    "-1.347,21",
    "XXX",
    "24-02-2020",
    "09:00",
    "ILLIMITY BANK",
    "IT0005359192",
    "MIL",
    "MTAA",
    "50",
    "10,13",
    "XXX",
    "-506,50",
    "XXX",
    "-506,50",
    "XXX",
    "-0,75",
    "XXX",
    "-507,25",
    "XXX",
    "12-02-2020",
    "09:04",
    "FINECOBANK",
    "IT0000072170",
    "MIL",
    "MTAA",
    "4",
    "11,025",
    "XXX",
    "-44,10",
    "XXX",
    "-44,10",
    "XXX",
    "-0,52",
    "XXX",
    "-44,62",
    "XXX",
    "30-01-2020",
    "16:36",
    "NOVARTIS AG",
    "CH0012005267",
    "SWX",
    "XVTX",
    "4",
    "92,50",
    "CHF",
    "-370,00",
    "CHF",
    "-346,25",
    "XXX",
    "1,0675",
    "-4,20",
    "XXX",
    "-350,45",
    "XXX",
    "02-12-2019",
    "16:00",
    "SPDR RUSSELL 2000 US",
    "SMALL CAP UCITS ETF",
    "IE00BJ38QD84",
    "MIL",
    "MTAA",
    "15",
    "40,20",
    "XXX",
    "-603,00",
    "XXX",
    "-603,00",
    "XXX",
    "-2,23",
    "XXX",
    "-605,23",
    "XXX",
    "15-11-2019",
    "10:02",
    "ERG",
    "IT0001157020",
    "MIL",
    "MTAA",
    "9",
    "18,85",
    "XXX",
    "-169,65",
    "XXX",
    "-169,65",
    "XXX",
    "-0,60",
    "XXX",
    "-170,25",
    "XXX",
    "15-11-2019",
    "09:31",
    "L&G BATTERY VALUE-CHAIN",
    "UCITS ETF",
    "IE00BF0M2Z96",
    "MIL",
    "MTAA",
    "40",
    "7,734",
    "XXX",
    "-309,36",
    "XXX",
    "-309,36",
    "XXX",
    "-2,12",
    "XXX",
    "-311,48",
    "XXX",
    "07-11-2019",
    "12:49",
    "POSTE ITALIANE",
    "IT0003796171",
    "MIL",
    "AQXE",
    "19",
    "10,80",
    "XXX",
    "-205,20",
    "XXX",
    "-205,20",
    "XXX",
    "-0,62",
    "XXX",
    "-205,82",
    "XXX",
    "07-11-2019",
    "12:21",
    "ENEL .",
    "IT0003128367",
    "MIL",
    "MTAA",
    "50",
    "6,80",
    "XXX",
    "-340,00",
    "XXX",
    "-340,00",
    "XXX",
    "-0,70",
    "XXX",
    "-340,70",
    "XXX",
    "31-10-2019",
    "13:54",
    "NEXI",
    "IT0005366767",
    "MIL",
    "MTAA",
    "70",
    "9,40",
    "XXX",
    "-658,00",
    "XXX",
    "-658,00",
    "XXX",
    "-0,88",
    "XXX",
    "-658,88",
    "XXX",
    "DEGIRO B.V. è registrata come",
    "Società d’Investimento presso l’Autorità",
    "dei Mercati Finanziari olandese.",
    "DEGIRO B.V.",
    "Amstelplein 1",
    "1096 HA Amsterdam",
    "E",
    "xxxxxxx@xxxxxx.xx",
    "I",
    "www.degiro.it",
    "Pagina 2 /",
    "2",
    "Estratto conto 2021-03-16"
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #2149: [Degiro] Degiro Italien
## Document
<details><summary>Portfolio.pdf</summary><p>

```json
[
  [
    "Xxxxxx XXXXX XXXXX",
    "Xxx xxx Xxxxxx 0",
    "00000 Xxxxxxxxx-Xxxxxx",
    "Xxxxx",
    "Xxxx Xxxxxx:",
    "*******xxx",
    "(000000000)",
    "Panoramica Portafoglio al 22-03-2021",
    "Prodotto",
    "Codice",
    "Quantità",
    "Ultimo",
    "Valore",
    "Valore in EUR",
    "CASH & CASH FUND & FTX CASH (EUR)",
    "EUR 73.96",
    "73,96",
    "BERKSHIRE HATHAWAY INC",
    "US0846707026",
    "2",
    "250,99",
    "USD 501.98",
    "422,42",
    "DEUTSCHE BANK AG",
    "DE0005140008",
    "45",
    "10,60",
    "EUR 476.91",
    "476,91",
    "FRESENIUS SE & CO KGAA",
    "DE0005785604",
    "12",
    "36,17",
    "EUR 434.04",
    "434,04",
    "ISHARES MSCI EM A",
    "IE00B4L5YC18",
    "27",
    "37,25",
    "EUR 1005.75",
    "1.005,75",
    "ISHARES MSCI WOR A",
    "IE00B4L5Y983",
    "32",
    "64,30",
    "EUR 2057.60",
    "2.057,60",
    "LEONARDO SPA",
    "IT0003856405",
    "50",
    "7,57",
    "EUR 378.60",
    "378,60",
    "VOLKSWAGEN AG",
    "DE0007664039",
    "1",
    "221,45",
    "EUR 221.45",
    "221,45",
    "Valore totale Portafoglio",
    "EUR 5070.73",
    "DEGIRO B.V. è registrata come",
    "Società d’Investimento presso l’Autorità",
    "dei Mercati Finanziari olandese.",
    "DEGIRO B.V.",
    "Amstelplein 1",
    "1096 HA Amsterdam",
    "E",
    "xxxxxxx@xxxxxx.xx",
    "I",
    "www.degiro.it",
    "Pagina 1 /",
    "1",
    "Estratto conto 2021-03-22"
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #2202: [Degiro] Es konnte kein Italienisches File Importiert werden
## Document
<details><summary>Transactions.pdf</summary><p>

```json
[
  [
    "Xxxxxx XXXXX XXXXXX",
    "Xx. Xxxxxxx 00 Xxxxx",
    "00000 Xxxxxxxxxx",
    "Xxxxx",
    "Xxxx Xxxxxx:",
    "*****Xxx",
    "(000000000)",
    "Operazioni da 19-02-2021 a 21-03-2021",
    "Data",
    "Ora",
    "Prodotto",
    "ISIN",
    "Borsa di",
    "Borsa",
    "Quantit",
    "à",
    "Quotazione",
    "Valore locale",
    "Valore",
    "Tasso",
    "di",
    "Commission",
    "e",
    "Totale",
    "18-03-2021",
    "09:05",
    "HSBC MSCI WORLD",
    "IE00B4X9L533",
    "EPA",
    "XPAR",
    "1",
    "23,728",
    "EUR",
    "-23,73",
    "EUR",
    "-23,73",
    "EUR",
    "-23,73",
    "EUR",
    "18-03-2021",
    "09:05",
    "HSBC MSCI WORLD",
    "IE00B4X9L533",
    "EPA",
    "XPAR",
    "2",
    "23,728",
    "EUR",
    "-47,46",
    "EUR",
    "-47,46",
    "EUR",
    "-47,46",
    "EUR",
    "18-03-2021",
    "09:05",
    "HSBC MSCI WORLD",
    "IE00B4X9L533",
    "EPA",
    "XPAR",
    "16",
    "23,728",
    "EUR",
    "-379,65",
    "EUR",
    "-379,65",
    "EUR",
    "-379,65",
    "EUR",
    "18-03-2021",
    "09:05",
    "ISHARES MSCI EM A",
    "IE00B4L5YC18",
    "EAM",
    "XAMS",
    "5",
    "37,39",
    "EUR",
    "-186,95",
    "EUR",
    "-186,95",
    "EUR",
    "-186,95",
    "EUR",
    "19-02-2021",
    "09:05",
    "ISHARES MSCI EM A",
    "IE00B4L5YC18",
    "EAM",
    "XAMS",
    "5",
    "38,955",
    "EUR",
    "-194,78",
    "EUR",
    "-194,78",
    "EUR",
    "-194,78",
    "EUR",
    "DEGIRO B.V. è registrata come",
    "Società d’Investimento presso l’Autorità",
    "dei Mercati Finanziari olandese.",
    "DEGIRO B.V.",
    "Amstelplein 1",
    "1096 HA Amsterdam",
    "E",
    "xxxxxxx@xxxxxx.xx",
    "I",
    "www.degiro.it",
    "Pagina 1 /",
    "1",
    "Estratto conto 2021-03-21"
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by SirGibihm**
Adding support for Depot statements
- Added Testcases:
  - Fixes #1742
- Checked locally using localhost:5000
- Removed bugs from transasction logs #1705 
  - Fixes #1705 

edit @dredav
- Fixes #2149
- Fixes #2148
- Fixes #2202

### CODE REVIEW

**Comment by dredav** [comment]
@SirGibihm Is this PR ready for review?

**Comment by SirGibihm** [comment]
> 
> 
> @SirGibihm Is this PR ready for review?

I am still working on some more degiro issues to be integrated within this PR so no, not yet

File: src/brokers/degiro.js [hunk 1]
```diff
@@ -29,80 +30,68 @@ class zeroSharesTransaction extends Error {
...
-  const isin = content[isinIdx];
-  const shares = Big(numberParser(content[isinIdx + 2 + offset]));
-  if (+shares === 0) {
+  let foreignCurrencyIndex;
+  const numberRegex = /^-?\d+(,\d+)?$/;
+  const isinIdx = findFirstIsinIndexInArray(content, index);
+  const sharesIdx = findFirstRegexIndexInArray(content, numberRegex, isinIdx);
+  const transactionEndIdx = findFirstRegexIndexInArray(
+    content,
+    /(DEGIRO B\.V\. ist als)|\d{2}-\d{2}-\d{4}/,
```
Comment (dredav) [inline src/brokers/degiro.js:40]: We should remove the German `ist als` to support the other language documents (e.g. IT https://github.com/tresorone/tresor-import/blob/58a456040ee29c24b2a3c3c42438e4a53a49e220/tests/brokers/__mocks__/degiro/transactionLog/buy_only_transactions_it.json#L102)

**Review by dredav — CHANGES_REQUESTED** [review]
Looks good, found one small issue with foreign languages.

**Comment by dredav** [comment]
Can you also resolve the conflicts?

**Comment by dredav** [comment]
@Zemmor Do you have a portfolio statement from degio in italian language that we can add as a seconds test case for this PR?

File: src/brokers/degiro.js [hunk 2]
```diff
@@ -148,32 +137,92 @@ const parseTransactionLog = pdfPages => {
...
 };
 
-export const canParseDocument = (pages, extension) => {
-  const firstPageContent = pages[0];
+const parseDepotStatement = pdfPages => {
+  const flattendPages = pdfPages.flat();
+  const dateline =
+    flattendPages[
+      flattendPages.findIndex(line =>
+        line.startsWith('Portfolioübersicht per ')
```
Comment (Zemmor) [inline src/brokers/degiro.js:145]: for Italian File:  "Panoramica Portafoglio al"

**Comment by Zemmor** [comment]
@dredav I found that Isssue in the project: https://github.com/tresorone/tresor-import/issues/2149

If you want another one, I can also Upload a portfolio statement file

**Review by dredav — APPROVED** [review]

**Pull request merged**